src/extract.rs

```diff
@@ -218,14 +225,81 @@ where
     }
 }
 
+impl FromRequest for Uri {
+    type Error = Infallible;
+    type Future = Ready<Result<Uri, Infallible>>;
+    type Config = ();
+
+    fn from_request(req: &HttpRequest, _: &mut Payload) -> Self::Future {
+        ok(req.uri().clone())
+    }
+}
+
+impl FromRequest for Version {
+    type Error = Infallible;
+    type Future = Ready<Result<Version, Infallible>>;
+    type Config = ();
+
+    fn from_request(req: &HttpRequest, _: &mut Payload) -> Self::Future {
+        ok(req.version())
+    }
+}
+
+impl FromRequest for Method {
+    type Error = Infallible;
+    type Future = Ready<Result<Method, Infallible>>;
+    type Config = ();
+
+    fn from_request(req: &HttpRequest, _: &mut Payload) -> Self::Future {
+        ok(req.method().clone())
+    }
+}
+
+impl FromRequest for ConnectionInfo {
+    type Error = Infallible;
+    type Future = Ready<Result<ConnectionInfo, Infallible>>;
+    type Config = ();
+
+    fn from_request(req: &HttpRequest, _: &mut Payload) -> Self::Future {
+        ok(req.connection_info().clone())
+    }
+}
+
+#[derive(Debug)]
+struct MissingPeerAddr;
+
+impl fmt::Display for MissingPeerAddr {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Missing peer address.")
+    }
+}
+
+impl ResponseError for MissingPeerAddr {}
+
+impl FromRequest for SocketAddr {
```

Comment:
I think i'd prefer a newtype struct for this `PeerAddr(SocketAddr)` since it _could_ be non-obvious what socket addr this refers to. Plus in general, it's easier to find these sorts of things in docs.